graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -296,6 +296,14 @@ public void setSource(final String source) {
     }
 
     public void addField(final String key, final Object value) {
+        addField(key, value, false);
+    }
+
+    private void addRequiredField(final String key, final Object value) {
+        addField(key, value, true);
+    }
+
+    private void addField(final String key, final Object value, final boolean isRequiredField) {
         // Don't accept protected keys. (some are allowed though lol)
         if (RESERVED_FIELDS.contains(key) && !RESERVED_SETTABLE_FIELDS.contains(key) || !validKey(key)) {
```

Comment:
While we are at it: why is the trimmed key used further down, while it is used non-trimmed here? Shouldn't we fix that to make it consistent?